view/frontend/web/instantsearch.js

```diff
@@ -207,408 +206,454 @@ requirejs(['algoliaBundle','Magento_Catalog/js/price-utils'], function(algoliaBu
 					'{{#isRefined}}<span class="cross-circle"></span>{{/isRefined}}' +
-					'<span class="{{cssClasses.count}}">{{#helpers.formatNumber}}{{count}}{{/helpers.formatNumber}}</span></a>' +
-					'</div>';
-				
-				return ['hierarchicalMenu', hierarchicalMenuParams];
-			}
-		};
-		
-		/** Add all facet widgets to instantsearch object **/
-		
-		window.getFacetWidget = function (facet, templates) {
-			
-			if (facet.type === 'priceRanges') {
-				delete templates.item;
-				
-				return ['priceRanges', {
-					container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-					attributeName: facet.attribute,
-					labels: {
-						currency: algoliaConfig.currencySymbol,
-						separator: algoliaConfig.translations.to,
-						button: algoliaConfig.translations.go
-					},
-					templates: templates,
-					cssClasses: {
-						root: 'facet conjunctive'
-					}
-				}];
-			}
-			
-			if (facet.type === 'conjunctive') {
-				var refinementListOptions = {
-					container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-					attributeName: facet.attribute,
-					limit: algoliaConfig.maxValuesPerFacet,
-					operator: 'and',
-					templates: templates,
-					sortBy: ['count:desc', 'name:asc'],
-					cssClasses: {
-						root: 'facet conjunctive'
-					}
-				};
-				
-				refinementListOptions = addSearchForFacetValues(facet, refinementListOptions);
-				
-				return ['refinementList', refinementListOptions];
-			}
-			
-			if (facet.type === 'disjunctive') {
-				var refinementListOptions = {
-					container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-					attributeName: facet.attribute,
-					limit: algoliaConfig.maxValuesPerFacet,
-					operator: 'or',
-					templates: templates,
-					sortBy: ['count:desc', 'name:asc'],
-					cssClasses: {
-						root: 'facet disjunctive'
+					'</a>';
+				hierarchicalMenuParams.panelOptions = {
+					templates: {
+						header: '<div class="name">' + (facet.label ? facet.label : facet.attribute) + '</div>',
 					}
 				};
-				
-				refinementListOptions = addSearchForFacetValues(facet, refinementListOptions);
-				
-				return ['refinementList', refinementListOptions];
-			}
-			
-			if (facet.type === 'slider') {
-				delete templates.item;
 
-				return ['rangeSlider', {
-					container: facet.wrapper.appendChild(createISWidgetContainer(facet.attribute)),
-					attributeName: facet.attribute,
-					templates: templates,
-					cssClasses: {
-						root: 'facet slider'
-					},
-					tooltips: {
-						format: function (formattedValue) {
-							return facet.attribute.match(/price/) === null ?
-								parseInt(formattedValue) :
-								priceUtils.formatPrice(formattedValue, algoliaConfig.priceFormat);
-						}
-					}
-				}];
+			return ['hierarchicalMenu', hierarchicalMenuParams];
+		}
+	};
+
+	/** Add all facet widgets to instantsearch object **/
+	window.getFacetWidget = function (facet, templates) {
```

Comment:
name is fairly awkward, and this is global. Where is it used?

Reply:
I didn't come up with the naming, I just moved the function
https://github.com/algolia/algoliasearch-magento-2/pull/838/files#diff-f45183f22871cde50da868482a4768dbL393
I'll remove it as global soon, I just need to make sure I didn't break anything :)